### PR TITLE
Cleanup testing utility

### PR DESCRIPTION
**Pull request by grofoli**
Some cleanup and refactoring, plus increased test coverage

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -61,6 +61,12 @@
             <version>1.7.30</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.awaitility</groupId>
```
Comment (cinemast) [inline pom.xml:65]: Not sure the extra dependency is worth the functionality.

File: src/main/java/os/failsafe/executor/utils/testing/AwaitableTaskExecutionListener.java [hunk 2]
```diff
@@ -20,20 +20,20 @@ public AwaitableTaskExecutionListener(long timeout, TimeUnit timeUnit) {
     }
 
     @Override
-    public void registered(String s, String taskId, String s1) {
+    public void registered(String name, String taskId, String parameter) {
```
Comment (cinemast) [inline src/main/java/os/failsafe/executor/utils/testing/AwaitableTaskExecutionListener.java:23]: Much better naming, Thanks!

File: src/test/java/os/failsafe/executor/TaskShould.java [hunk 3]
```diff
@@ -6,23 +6,20 @@
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 
-public class TaskShould {
+class TaskShould {
 
     @Test
     void return_false_on_cancel_if_task_is_not_cancelable_because_it_is_locked() {
-        Task task = new Task(new String("id"), "parameter", "name", LocalDateTime.now(), LocalDateTime.now(), null, 0L, null);
```
Comment (cinemast) [inline src/test/java/os/failsafe/executor/TaskShould.java:13]: Leftover from the refacotring, thanks!

File: pom.xml (hunk 1 above)
Comment (grofoli) [reply pom.xml:65]: Well, it is a test scoped dependency on purpose and awaitility has more features than a while loop. But this is just a suggestion of course.

File: src/test/java/os/failsafe/executor/MultipleNodesShould.java [hunk 4]
```diff
@@ -127,12 +126,8 @@ private long countFinishedTasks(TaskExecutionListener l) {
...
 
     private void awaitEmptyTaskTable() {
-        while (taskRepository.findAll().size() > 0) {
-            try {
-                Thread.sleep(10);
-            } catch (InterruptedException e) {
-                e.printStackTrace();
-            }
-        }
+        await().atMost(Duration.ofSeconds(1))
```
Comment (grofoli) [inline src/test/java/os/failsafe/executor/MultipleNodesShould.java:129]: You can simply write await().until(...). atMost and pollInterval are not mandatory. I just wanted to make it explicit.

**Pull request merged**